### PR TITLE
Compile and minify CSS using css-loader

### PR DESCRIPTION
**Pull request by ESWAT**
`style.css` now gets fed through webpack using `css-loader` and `extract-text-webpack-plugin` to solve some low-hanging fruit like local imports and minification. I’ve put the font declarations in a separate file to demonstrate the local import part.

I could have gone ahead and also add `postcss-loader` but figured to go one step at a time; can probably get a lot of mileage out of just being able to have multiple CSS files from source.

Related to #1869 and #1551.

*Note*: For some reason the sourcemap for the CSS gets generated and put in the `public/css` directory even if I tested with setting the `sourceMap: false` declaration for the `css-loader` options… if anyone has any ideas?

### CODE REVIEW

**Comment by xPaw** [comment]
Thanks for doing this. Have you look at the changes in webpack 4 in regards to CSS yet? It might be worthwhile to wait for webpack 4 before attempting this.

> For some reason the sourcemap for the CSS gets generated and put in the public/css directory

Is that a problem?

**Comment by ESWAT** [comment]
> Thanks for doing this. Have you look at the changes in webpack 4 in regards to CSS yet? It might be worthwhile to wait for webpack 4 before attempting this.
I have not but can test out what a similar webpack 3 config with these packages would need to upgrade to webpack 4.

> Is that a problem?
I guess not. This actually made me realize I should make a dev config for the CSS that doesn’t minimize the CSS.

**Comment by astorije** [comment]
Unfortunately, some of the Webpack plugins we use seem to stagnate and it might be a while before we can upgrade to Webpack v4 (unless we're okay with forks, if/when they're available).
@xPaw, wouldn't you prefer going forward with this and make progress on our end, and worry about Webpack v4 upgrade when we actually can?

File: webpack.config.js [hunk 1]
```diff
@@ -87,8 +105,8 @@ const config = {
 				from: "./client/themes/*",
 				to: "themes/[name].[ext]",
 			},
-			{ // TODO: Build css with postcss
-				from: "./client/css/*",
+			{
+				from: "./client/css/bootstrap.css",
```
Comment (xPaw) [inline webpack.config.js]: Can you just add that as `@import` (same with tooltips)

File: webpack.config.js [hunk 2]
```diff
@@ -20,6 +21,23 @@ const config = {
...
+				include: [
+					path.resolve(__dirname, "client"),
+				],
+				use: ExtractTextPlugin.extract([
+					{
+						loader: "css-loader",
+						options: {
+							allChunks: true,
+							importLoaders: 1,
+							minimize: process.env.NODE_ENV === "production" ? true : false,
```
Comment (xPaw) [inline webpack.config.js]: `? true : false` is unnecessary. It already evaluates to a boolean.

**Comment by xPaw** [comment]
@astorije My only worry is `ExtractTextPlugin`, which I believe is no longer necessary in v4, so I'd like to know how that's handled there.

**Comment by astorije** [comment]
Not sure if you've seen this or if it's any relevant, but: https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/701#issuecomment-374551300

**Comment by xPaw** [comment]
Gonna close this as I tackled it in #2117.

**Pull request closed without merging**